component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -586,6 +605,39 @@ public static void redirectToEnableTOTPReqPage(HttpServletRequest request, HttpS
         }
     }
 
+    /**
+     * Redirect the enableTOTP request page.
+     *
+     * @param request  The HttpServletRequest
+     * @param response The HttpServletResponse
+     * @param context  The AuthenticationContext
+     * @param skey     QR code claim
+     * @param runtimeParams runtime parameters
+     * @throws AuthenticationFailedException On error while getting value for enrolUserInAuthenticationFlow
+     */
+    public static void redirectToEnableTOTPReqPage(HttpServletRequest request, HttpServletResponse response,
+                                                   AuthenticationContext context, String skey,
+                                                   Map<String,String> runtimeParams)
+            throws AuthenticationFailedException {
+
+        if (isEnrolUserInAuthenticationFlowEnabled(context, runtimeParams)) {
+            String multiOptionURI = getMultiOptionURIQueryParam(request);
+            String queryParams = "sessionDataKey=" + context.getContextIdentifier() + "&authenticators=" +
+                    TOTPAuthenticatorConstants.AUTHENTICATOR_NAME + "&type=totp" + "&ske=" + skey + multiOptionURI;
+            String enableTOTPReqPageUrl =
+                    FrameworkUtils.appendQueryParamsStringToUrl(getEnableTOTPPage(context), queryParams);
+
+            try {
+                response.sendRedirect(enableTOTPReqPageUrl);
+            } catch (IOException e) {
+                throw new AuthenticationFailedException(
+                        "Error while redirecting the request to get enableTOTP " + "request page. ", e);
```

Comment:
```suggestion
                        "Error while redirecting the request to get enableTOTP request page. ", e);
```

Check on the wording as well